base/float.jl

```diff
@@ -447,7 +447,7 @@ Test whether a number value is a NaN, an indeterminate value which is neither an
 nor a finite number ("not a number").
 """
 isnan(x::AbstractFloat) = (x != x)::Bool
-isnan(x::Real) = false
+isnan(x::Number) = false
 
 isfinite(x::AbstractFloat) = x - x == 0
```

Comment:
How about `isfinite`? I think it is also not defined for `Number`? 

Also a test would be good so this does not get deleted by accident. 
```julia
struct N38232 <: Number end
@test isfinite(N38232())
@test !isnan(N38232())
```

Reply:
`isfinite` is not `true` by default. It is using `Base.decompose` for `Real` which is not exposed therefore I would leave the decision what to do it for a separate PR (fell free to open it; I need `isnan` for DataFrames.jl, as `NaN` has a special treatment in comparisons).

I will add the test you suggest - thank you.